map_tests/src/turns.rs

```diff
@@ -0,0 +1,71 @@
+use crate::import_map;
+use map_model::{LaneID, TurnType};
+
+// If this test gets fully fleshed out, there would be many more of these. This one means the
+// southern road, inbound to the intersection, the left lane of it.
+const S_IN_LEFT: LaneID = LaneID(2);
+
+#[test]
+fn test_left_turn_lane() {
+    let map = import_map(four_way());
+
+    // Assert the hardcoded ID is reasonable
+    assert_eq!("south", map.get_parent(S_IN_LEFT).get_name(None));
+    assert!(map.get_l(S_IN_LEFT).is_driving());
+
+    // TODO This is quite a weak assertion. I want to express that there's only the left turn from
```

Comment:
I can't imagine any form of this test that's easy to read or write, which means it's useless for actually quickly iterating on turn restrictions. :(

Maybe the goldenfile approach is what we need. Keep the fixed .osm, run conversion, print all the turn IDs in some text form, check it in. The text itself is kind of useless by itself, but when it changes, pop into the UI and manually inspect the intersection. Thoughts? I might spin this PR in that direction instead.

Reply:
Yeah, tests like you've written here are nice in that you can be really specific, but they are indeed hard to write and hard to read. 

I like what you've proposed with the golden file - just having a regression check could be really helpful, provided we don't have a lot of inexplicable false alarms.

It might be tricky to come up with a good diffable text format to check in, but we can iterate on that too.

Instead of just one larger map with a bunch of different features, it might be helpful to, over time , build up a menagerie of smaller "unit" maps which test certain canonical cases, especially if there are certain troublesome patterns. 

This infrastructure might also be helpful while working on certain thorny map import problems. Rather than launching the whole game with a real scenario, which is a quite slow feedback loop for me, instead crafting a representative unit map to work on/view might be much faster to rev on.

Reply:
> It might be tricky to come up with a good diffable text format to check in, but we can iterate on that too.

How do you feel about `data/route_goldenfiles/downtown.txt` as it is? I'll try something similar, listing (from lane, to lane, turn type) maybe.

> smaller "unit" maps

Agreed! The thing that would make this really attractive would be a fast way to select some stuff from a normal map and export the "essence" of it to a minimal test map.